docs/docsite/rst/dev_guide/developing_program_flow_modules.rst

```diff
@@ -378,79 +378,74 @@ Internal arguments
 
-The internal arguments listed here are global. If you need to add a local internal argument to a custom module, create an action plugin for that specific module - see ``_original_basename`` in the `copy action plugin <https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/copy.py#L329>`_ for an example.
 
 _ansible_no_log
 ^^^^^^^^^^^^^^^
 
-Boolean. Set to True whenever a parameter in a task or play specifies ``no_log``. Any module that calls :py:meth:`AnsibleModule.log` handles this automatically. If a module implements its own logging then
-it needs to check this value. To access in a module, instantiate an
-``AnsibleModule`` and then check the value of :attr:`AnsibleModule.no_log`.
+Format: <bool>
+
+Set to ``True`` whenever an argument in a task or play specifies ``no_log``. Any module that calls the :py:meth:`AnsibleModule.log` function handles this action automatically. If you have a module that implements its own logging then you need to check the value of ``_ansible_no_log``. To access ``_ansible_no_log`` in a module, instantiate the ``AnsibleModule`` utility and then check the value of :attr:`AnsibleModule.no_log`.
 
 .. note::
-    ``no_log`` specified in a module's argument_spec is handled by a different mechanism.
+    ``no_log`` you specified in the ``argument_spec`` specification of a module is handled by a different mechanism.
+
 
 _ansible_debug
-^^^^^^^^^^^^^^^
+^^^^^^^^^^^^^^
+
+Format: <bool>
+
+Operates verbose logging and logging of external commands that a module executes. If the module uses the :py:meth:`AnsibleModule.debug` function rather than the :py:meth:`AnsibleModule.log` function then the messages are only logged if you set the ``_ansible_debug`` argument to ``True``. To activate the setting, add the ``debug: True`` key to the :file:`ansible.cfg` configuration file. Alternatively, you can set the environment variable :envvar:`ANSIBLE_DEBUG`. To access ``_ansible_debug`` in a module, instantiate the ``AnsibleModule`` utility and access :attr:`AnsibleModule._debug`.
 
-Boolean. Turns more verbose logging on or off and turns on logging of
-external commands that the module executes. If a module uses
-:py:meth:`AnsibleModule.debug` rather than :py:meth:`AnsibleModule.log` then
-the messages are only logged if ``_ansible_debug`` is set to ``True``.
-To set, add ``debug: True`` to :file:`ansible.cfg` or set the environment
-variable :envvar:`ANSIBLE_DEBUG`. To access in a module, instantiate an
-``AnsibleModule`` and access :attr:`AnsibleModule._debug`.
 
 _ansible_diff
-^^^^^^^^^^^^^^^
+^^^^^^^^^^^^^
+
+Format: <bool>
+
+With this parameter you can configure your module to show a unified diff of changes that will be applied to the templated files. To activate the setting, use the ``--diff`` command-line option. To access ``_ansible_diff`` in a module, instantiate the ``AnsibleModule`` utility and access :attr:`AnsibleModule._diff`.
```

Comment:
there is also the `diff` keyword, most setting are 'settable' in multple ways and at different scopes.

Reply:
You mean that the `diff` keyword is available for the `_ansible_diff` argument? Should we mention that keyword here with some description?

Reply:
the keyword is already described in the keywords page, but my point is not only about this specific setting, most of these keys result from multiple levels of configuration from global to per play/role/taks/host, i would link to the source of the configuration (for example, remote_tmp configuration in shell plugins for _remote_tmp)

Reply:
also, the config sources will already have more information like 'type', 'default', 'choices', etc

Reply:
`_ansible_socket` is created in https://docs.ansible.com/ansible/latest/reference_appendices/config.html#persistent-control-path-dir

`_ansible_tmpdir` is the connection plugin's shell's `tmpdir` attribute. It's set here https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/__init__.py#L503. Depending on the circumstance the tmp dir will be created in one of these: 
* [DEFAULT_LOCAL_TMP](https://docs.ansible.com/ansible/latest/reference_appendices/config.html#default-local-tmp) 
*  [remote_tmp](https://github.com/ansible/ansible/blob/4d984613f5e16e205434cdf7290572e62b40bf62/lib/ansible/plugins/doc_fragments/shell_common.py#L12-L21) shell option
* [system_tmpdirs](https://github.com/ansible/ansible/blob/4d984613f5e16e205434cdf7290572e62b40bf62/lib/ansible/plugins/doc_fragments/shell_common.py#L35-L52) shell option

Reply:
@bcoca @s-hertel - is this set of comments resolved by @JaroslavKlech recent changes here? Hard for me to tel...

Reply:
@samccann Not for this specific option, but for some of the others. The keyword `diff` https://docs.ansible.com/ansible/latest/reference_appendices/playbooks_keywords.html and configuration entry https://docs.ansible.com/ansible/latest/reference_appendices/config.html#diff-always should be mentioned here.